lib/gems/pending/util/object_storage/miq_s3_storage.rb

```diff
@@ -0,0 +1,101 @@
+require 'util/miq_object_storage'
+
+class MiqS3Storage < MiqObjectStorage
+  attr_reader :bucket_name
+
+  def self.uri_scheme
+    "s3".freeze
+  end
+
+  def self.new_with_opts(opts)
+    new(opts.slice(:uri, :username, :password, :region))
+  end
+
+  def initialize(settings)
+    super(settings)
+
+    # NOTE: This line to be removed once manageiq-ui-class region change implemented.
+    @settings[:region] ||= "us-east-1"
```

Comment:
Since we're here anyway ... @jerryk55 can we resolve this `NOTE:`?

Reply:
Okay, but if we're keeping it, then we can probably drop the comment. Waiting on your input here @jerryk55